### DISCUSSION TITLE
Saved filters with custom filter options in a plugin

### DISCUSSION THREAD

**Discussion by dreng** [netbox-community/netbox#12397] [Help Wanted!]
Hi developers,

I am co-developer of the plugin "netbox-topology-views" and I am struggling with a problem related to Saved Filters. I hope there is someone here who can bring us a decisive step forward.

The problem is that in addition to normal filters and custom fields, we have some options that can be activated and deactivated. The options end up as JSON in a Saved Filter, but are not taken into account when the Saved Filter is applied. After looking at the NetBox code, I am unsure if this is even possible. Is the problem with the plugin or would something need to be changed in NetBox?

This is what the filter pane looks like:
![topo_options](https://user-images.githubusercontent.com/20901110/235304173-33661416-fa95-4093-90f1-eb79923d2fdc.png)

And here's an example of a Saved Filter JSON (this is an old screenshot and therefore does not correspond to the first screenshot):
![topo_json](https://user-images.githubusercontent.com/20901110/235304317-904c5fe2-d062-4260-ba64-490f1c681f82.png)

See also https://github.com/mattieserver/netbox-topology-views/issues/236

Thank you very much in advance for your contribution!

**Comment by dreng** [comment]
Hi,

Unfortunately we have not made any progress in answering this question ourselves. It would be great if someone from the team could comment on this. @jeremystretch , @DanSheps , @arthanson , anyone?

**Reply by DanSheps** [reply]
This would likely be a plugin problem.

**Comment by dreng** [comment]
Thanks for your answer! The red marked options in the screenshot above do not belong to any model. Might this be a problem?
